### PR TITLE
modefied the pp build for ckpt adaptation

### PR DESCRIPTION
**Pull request by Gy-Lu**
Our former pipeline build is not suitable for ckpt.
I have fixed it.

### CODE REVIEW

File: colossalai/builder/pipeline.py [hunk 1]
```diff
@@ -250,9 +250,15 @@ def build_pipeline_model(layers: nn.Sequential, num_chunks: int = 1, verbose: bo
     pipeline_parallel_size = gpc.get_world_size(ParallelMode.PIPELINE)
     pipeline_rank = gpc.get_local_rank(ParallelMode.PIPELINE)
     partitions = partition_uniform(len(layers), pipeline_parallel_size, num_chunks)
-    module_list = []
+    layer_marks = []
     for start, end in partitions[pipeline_rank]:
-        module_list.append(nn.Sequential(*layers[start:end]))
+        layer_marks += [i for i in range(start, end)]
+    module_list = []
+    for i, _ in enumerate(layers):
```
Comment (ver217) [inline colossalai/builder/pipeline.py:257]: You can just use `for i in range(layers)`.

File: colossalai/builder/pipeline.py [hunk 2]
```diff
@@ -262,4 +268,4 @@ def build_pipeline_model(layers: nn.Sequential, num_chunks: int = 1, verbose: bo
                 log_str += f'===== chunk={chunk}, layer=[{start}-{end}] =====\n'
                 log_str += '\n'.join([str(layer) for layer in layers[start:end]]) + '\n'
             logger.info(log_str, ranks=[0])
-    return nn.ModuleList(module_list) if len(module_list) > 1 else module_list[0]
+    return nn.Sequential(*tuple(module_list)) if len(module_list) > 1 else module_list[0]
```
Comment (ver217) [inline colossalai/builder/pipeline.py:271]: `ModuleList` is used for interleaved pipeline. Each module in this `ModuleList` is a `Sequential` module. Why do you use `Sequential` here? Actually I think they shouldn't be sequential.

File: colossalai/builder/pipeline.py (hunk 1 above)
Comment (Gy-Lu) [reply colossalai/builder/pipeline.py:257]: I am not sure. I need the no. of the layer. `range(layers)` seems to provide the layer name?

File: colossalai/builder/pipeline.py (hunk 2 above)
Comment (Gy-Lu) [reply colossalai/builder/pipeline.py:271]: I forget to consider the interleaved pipeline. Seems `ModuleList` is fine, I'll test it.

Comment (Gy-Lu) [reply colossalai/builder/pipeline.py:271]: I reconsider it. I think the interleaved pipeline is supported, for we use `torch.nn.Identity`. `Sequntial` is ok as well.

Comment (Gy-Lu) [reply colossalai/builder/pipeline.py:271]: I am totally wrong. It needs to be redesigned.

**Pull request closed without merging**